lampod/src/liquidity.rs

```diff
@@ -0,0 +1,283 @@
+    pub ctlv_exiry: Option<u32>,
+}
+
+pub struct LampoLiquidityManager {
+    // lampod: Arc<LampoDaemon>,
+    lampo_liquidity: Arc<LampoLiquidity>,
+    lampo_conf: Arc<LampoConf>,
+    // TODO: We can't use Arc here as we are modifying data. Fix this.
+    lsp_provider: Option<LSPProvider>,
+    channel_manager: Arc<LampoChannelManager>,
+    keys_manager: Arc<LampoKeysManager>,
+}
+
+impl LampoLiquidityManager {
+    pub fn new(
+        liquidity: Arc<LampoLiquidity>,
+        conf: Arc<LampoConf>,
+        provider: Option<LSPProvider>,
+        channel_manager: Arc<LampoChannelManager>,
+        keys_manager: Arc<LampoKeysManager>,
+    ) -> Self {
+        Self {
+            lampo_liquidity: liquidity,
+            lampo_conf: conf,
+            lsp_provider: provider,
+            channel_manager,
+            keys_manager,
+        }
+    }
+
+    // Behaving as a client
+    pub fn new_liquidity_consumer(
+        liquidity: Arc<LampoLiquidity>,
+        conf: Arc<LampoConf>,
+        channel_manager: Arc<LampoChannelManager>,
+        keys_manager: Arc<LampoKeysManager>,
+        address: SocketAddress,
+        node_id: PublicKey,
+        token: Option<String>,
+    ) -> Self {
+        // FIXME: Implement a new function for this
+        let lsp_provider = LSPProvider {
+            addr: address,
+            node_id,
+            token,
+            opening_params: None,
+            ctlv_exiry: None,
+            scid: None,
+        };
+        Self {
+            lampo_liquidity: liquidity,
+            lampo_conf: conf,
+            channel_manager,
+            lsp_provider: Some(lsp_provider),
+            keys_manager,
+        }
+    }
+
+    pub fn get_events(&self) {
+        self.lampo_liquidity.get_and_clear_pending_events();
+    }
+
+    pub async fn listen(&mut self) {
+        match self.lampo_liquidity.next_event_async().await {
+            Event::LSPS0Client(..) => todo!(),
+            Event::LSPS2Client(LSPS2ClientEvent::OpeningParametersReady {
+                counterparty_node_id,
+                opening_fee_params_menu,
+                ..
+            }) => {
+                // This event is called when the LSP server provides us with
+                // all the OpeningChannelParams in response of get_info request
+                if &self.lsp_provider.as_ref().unwrap().node_id != &counterparty_node_id {
+                    log::info!("Recieved Unknown OpeningParametersReady event")
+                }
+
+                // TODO: Handle this in a better way as we can get new opening_params from a
+                // LSP if it fails to responds within a certain time
+                if self.lsp_provider.as_ref().unwrap().opening_params.is_some() {
+                    log::info!("We already have some params inside lsp_provider")
+                }
+
+                self.lsp_provider.as_mut().unwrap().opening_params = Some(opening_fee_params_menu)
+            }
```

Comment:
@vincenzopalazzo For example this event occurs when we call the function `client_request_opening_params`, this contains the information about the a LSP fee rates. So, how do we wait for this event to happen and then proceed with the program flow.

Reply:
still your question is unclear, are you looking some code example that should be included inside your question?

Reply:
Idk I should look inside the code

Reply:
Okay I got a work around